library/functions.lisp

```diff
@@ -142,5 +143,30 @@
   (define-instance (Functor (Arrow :a))
     (define map compose)))
 
+;;;
+;;; Bracket pattern
+;;;
+
+(cl:defmacro %unwind-protect (obj exit thunk)
+  "A wrapper on `cl:unwind-protect.`"
+  `(cl:let ((output (cl:gensym "OUTPUT")))
```

Comment:
I think this is gensymming wrong.

```
(let ((output (gensym "OUTPUT")))
  `( ... ,output))
```

The let should be outside the backquote, and instances of output should be unquoted.